### PR TITLE
Lesson 7 dragla marius

### PR DESCRIPTION
**Pull request by mariusamsarius**

### CODE REVIEW

File: lesson_7/lesson7_decorators.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# "GREETINGS WORLD!"
+
+def uppercase(fnc):
+    def inner_fnc(*args):
+        initial_str = fnc(*args)
+        uppercase_str = initial_str.upper()
+        return uppercase_str
```
Comment (lorena91) [inline lesson_7/lesson7_decorators.py]: As a shorter variant i would return upper function in a single line:
`return fnc(*args).upper()`

**Review by lorena91 — COMMENTED** [review]
Looks good to me :+1:

File: lesson_7/lesson7_decorators.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+# Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(fnc):
+
+   def wrapper(*args, **kwargs):
+        try:
+            return fnc(*args, **kwargs)
+        except Exception as e:
+            print("Division by zero error")
```
Comment (mariuspodean) [inline lesson_7/lesson7_decorators.py]: Would this still be a valid error if I try the following:
```
divide(10, 'd')
```
?

File: lesson_7/lesson7_decorators.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+# Create a decorator called register that will update a list called print_registry with all the decorated functions names.
+
+print_registry = []
+
+
+def register(fnc):
+    print_registry.append(fnc.__name__)
+    return fnc
```
Comment (mariuspodean) [inline lesson_7/lesson7_decorators.py]: You are not using an inner function; why is that? Please justify your answer

**Review by mariuspodean — CHANGES_REQUESTED** [review]
The folder structure should be:
lesson_7/name_surname/

**Comment by mariusamsarius** [comment]
> The folder structure should be:
> lesson_7/name_surname/

Not sure how to change the folder structure on Git as folders and files from lesson 7 have somehow disappeared from my local directory after doing the push. Do I need to make a pull from my branch (lesson_7_dragla_marius), change the folder structure and then do the commits and push?

File: lesson_7/lesson7_decorators.py (hunk 1 above)
Comment (mariusamsarius) [reply lesson_7/lesson7_decorators.py]: I thought explicit would be better than implicit.. :) but I have now changed the code into a single line

File: lesson_7/lesson7_decorators.py (hunk 2 above)
Comment (mariusamsarius) [reply lesson_7/lesson7_decorators.py]: changed the error message to "Division error"

File: lesson_7/lesson7_decorators.py (hunk 3 above)
Comment (mariusamsarius) [reply lesson_7/lesson7_decorators.py]: is it mandatory to use an inner function since the decorator doesn't do any changes to the decorated functions?

Comment (mariuspodean) [reply lesson_7/lesson7_decorators.py]: Nope, it's not; just wanted to know your argument for this

**Review by mariuspodean — APPROVED** [review]

**Pull request merged**